services/wireguard/endpoint/create_linux.go

```diff
@@ -40,12 +52,8 @@ func NewConnectionEndpoint(ipResolver ip.Resolver, resourceAllocator *resources.
 	}, nil
 }
 
-// GeneratePrivateKey creates new wireguard private key
-func GeneratePrivateKey() (string, error) {
-	return kernelspace.GeneratePrivateKey()
-}
-
-// PrivateKeyToPublicKey generates wireguard public key from private key
-func PrivateKeyToPublicKey(key string) (string, error) {
-	return kernelspace.PrivateKeyToPublicKey(key)
+func isKernelSpaceSupported() bool {
+	err := utils.SudoExec("ip", "link", "add", "iswgsupported", "type", "wireguard")
+	utils.SudoExec("ip", "link", "del", "iswgsupported")
```

Comment:
I think we should explicitly state that we understand that an error is returned, but we don't care.
```suggestion
	_ = utils.SudoExec("ip", "link", "del", "iswgsupported")
```